variation/tokenizers/protein_deletion.py

```diff
@@ -1,76 +1,106 @@
-        if input_string is None:
-            return None
+        """Return token that match the input string
 
-        self.parts = {
+        :param str input_str: Input string to tokenize
+        :return: `ProteinDeletionToken` if a match is found, else `None`
+        """
+        parts = {
             "used_one_letter": False,
             "token": input_string,
             "input_string": input_string,
             "match_type": TokenMatchType.UNSPECIFIED.value,
             "start_aa_del": None,
             "start_pos_del": None,
             "end_aa_del": None,
-            "end_pos_del": None
+            "end_pos_del": None,
+            "deleted_aa": None
         }
 
-        input_string = str(input_string).lower()
+        input_str_l = str(input_string).lower()
 
-        if "c." in input_string or "g." in input_string:
+        if input_str_l.startswith(("c.", "g.")):
             return None
 
-        if input_string.startswith("p."):
-            input_string = input_string[2:]
+        if input_str_l.startswith("p."):
+            input_str_l = input_str_l[2:]
 
-        if input_string.startswith("(") and input_string.endswith(")"):
-            input_string = input_string[1:-1]
+        if input_str_l.startswith("(") and input_str_l.endswith(")"):
+            input_str_l = input_str_l[1:-1]
 
-        if not input_string.endswith("del"):
+        match = self.splitter.match(input_str_l)
+        if not match:
             return None
 
-        parts = input_string.split("del")
-        self._get_parts(parts)
+        params = match.groupdict()
+        one_letter_aa = False  # Whether or not 1 letter AA code was used
+
+        parts["start_aa_del"] = params["start_aa"]
+        parts["start_pos_del"] = params["start_pos"]
+        parts["end_aa_del"] = params["end_aa"]
+        parts["end_pos_del"] = params["end_pos"]
+        parts["deleted_aa"] = params["deleted_aa"]
+
+        # This ensures that start/end/deleted AA use consistent 1 or 3 AA codes
+        if len(parts["start_aa_del"]) == 1:
+            one_letter_aa = True
+
+        # Validate start and end deleted amino acids (if given)
+        for key in ["start_aa_del", "end_aa_del"]:
+            aa = parts[key]
+            if aa:
+                if one_letter_aa:
+                    try:
+                        parts[key] = aa1_to_aa3(aa.upper())
+                    except KeyError:
+                        return None
+                else:
+                    if aa.capitalize() not in aa3_to_aa1_lut:
+                        return None
+                    else:
+                        parts[key] = aa.capitalize()
+
+        # Validate deleted amino acid sequence
+        if parts["deleted_aa"]:
+            deleted_aa = ""
+            if one_letter_aa:
+                for i in range(len(parts["deleted_aa"])):
+                    aa = parts["deleted_aa"][i:i + 1]
+                    assert len(aa) == 1
```

Comment:
maybe it already is somewhere, but if not, I think this AssertionError should be caught (or should return None if len > 1 or something)

Reply:
(if it's handled somewhere already, I can go ahead and approve)

Reply:
Oh good catch. This was something that needs to be removed